hbase-assembly/src/main/assembly/hadoop-three-compat.xml

```diff
@@ -261,6 +262,14 @@
         <include>io.opentelemetry.javaagent:*</include>
       </includes>
     </dependencySet>
+
+    <!-- Adds junit libs to lib/test -->
+    <dependencySet>
+      <outputDirectory>lib/test</outputDirectory>
```

Comment:
Will we load the libraries under this package?

And better also include hamcrest here.

Reply:
Yes, these libraries would be packed under a "lib\test" folder, and would not be added to hbase classpath automatically (the hbase script, by default, puts "lib\*" on the CP).

Reply:
I mean we should add these libraries?

Reply:
So how do we setback when executing IntegrationTestIngest? Manually?

Reply:
Yes, this could be easily done as `HBASE_CLASSPATH="$HBASE_HOME/lib/tests/*" hbase org.apache.hadoop.hbase.IntegrationTestIngest -m slowDeterministic`, since we already provide the lib within the tarball. We could update the ref guide section accordingly.

Reply:
I wonder what is the problem if we always add these jars to our classpath? At lease when running tools(not daemon services)?

Reply:
Ehh, usually it's been a thing that causes people to ask questions rather than something actually breaking. Apparently others think that we shouldnt' have this on the $CLASSPATH given they removed them.

Reply:
For me, it is a security and release hygiene issue to ship test dependencies in the runtime. It is problematic for me that some of our user utilities have runtime dependency on test jars. Only mildly related: I'm curious how much extra space those test dependencies consume in our release artifacts.

Short of the restructuring that I suggested, I am okay with this approach of isolating these test dependencies in their own directory and conditionally including them at runtime. If there are classes that we know require them, our `bin/hbase` script can intelligently add them to the classpath for those classes. Doing so will preserve backward compatibility while also keeping the extra jars out of the  daemon services' classpaths.

Reply:
Ok, let me add it to the CP of the tools that require it, in `bin/hbase` script.